pyterrier/terrier/retriever.py

```diff
@@ -301,13 +314,19 @@ def __setstate__(self, d):
     def _retrieve_one(self, row, input_results=None, docno_provided=False, docid_provided=False, scores_provided=False):
         rank = FIRST_RANK
         qid = str(row.qid)
-        query = row.query
-        if len(query) == 0:
-            warn("Skipping empty query for qid %s" % qid)
-            return []
 
-        srq = self.manager.newSearchRequest(qid, query)
-        
+        # row is a namedtuple, whose fields are exposed in _fields
+        query_toks = 'query_toks' in row._fields
+        if query_toks:
+            query = ''
```

Comment:
`query = _querytoks2matchop(row.query_toks)`?

Reply:
Oh, I see it's done below. I think it's clearer if you do it up here. Or at least give a comment saying that it's done below.

Reply:
aim is just to detect query_toks mode. Probably the variable should be "query_toks_present" to be clear its a bool flag.